### PR TITLE
Modify rule S5655: LaYC format

### PR DESCRIPTION
**Pull request by joke1196**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5655/python/rule.adoc [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
+This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations.
 
+== Why is this an issue?
 
-This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations. It also checks argument types for builtin functions.
+The CPython interpreter does not check arguments type when functions are called. 
+However a function can express the type it expects for each argument in its documentation or by using https://www.python.org/dev/peps/pep-0484/[Type Hints]. 
+Calling a function with an argument of a different type than the one expected can lead to bugs as well as maintainability issues. 
+For example the code could work right now, but it can fail later when APIs evolve or when type checks are added (ex: with ``++isinstance++``).
```
Comment (joke1196) [inline rules/S5655/python/rule.adoc:8]: Not 100% this is clear enough. I wasn't sure if I should go more into details as well.

Comment (guillaume-dequenne-sonarsource) [inline rules/S5655/python/rule.adoc:8]: What about something like:
While the code may initially work as intended, not respecting the contract of an API may lead to bugs later when its implementation evolves or when type checks are added (ex: with `isinstance`). 
?

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
Looks good, left a minor rewording suggestion, feel free to ignore it though.

File: rules/S5655/python/rule.adoc [hunk 2]
```diff
@@ -1,42 +1,48 @@
-== Why is this an issue?
-
-The CPython interpreter does not check arguments type when functions are called. However a function can express the type it expects for each argument in its documentation or by using https://www.python.org/dev/peps/pep-0484/[Type Hints]. Calling such a function with an argument of a different type can easily create a bug. Even if it works right now it can fail later when APIs evolve or when type checks are added (ex: with ``++isinstance++``).
+This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations.
 
+== Why is this an issue?
 
-This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations. It also checks argument types for builtin functions.
+The CPython interpreter does not check arguments type when functions are called.
```
Comment (jean-jimbo-sonarsource) [inline rules/S5655/python/rule.adoc:5]: Grammar suggestion: "The CPython interpreter does not check an argument's type when functions are called."

File: rules/S5655/python/rule.adoc [hunk 3]
```diff
@@ -1,42 +1,48 @@
-== Why is this an issue?
-
-The CPython interpreter does not check arguments type when functions are called. However a function can express the type it expects for each argument in its documentation or by using https://www.python.org/dev/peps/pep-0484/[Type Hints]. Calling such a function with an argument of a different type can easily create a bug. Even if it works right now it can fail later when APIs evolve or when type checks are added (ex: with ``++isinstance++``).
+This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations.
 
+== Why is this an issue?
 
-This rule raises an issue when a function or method is called with an argument of a different type than the one described in its type annotations. It also checks argument types for builtin functions.
+The CPython interpreter does not check arguments type when functions are called. 
+However a function can express the type it expects for each argument in its documentation or
```
Comment (jean-jimbo-sonarsource) [inline rules/S5655/python/rule.adoc:6]: Follow however with a comma: "However, a function..."

**Review by jean-jimbo-sonarsource — APPROVED** [review]
Added some grammatical suggestions

File: rules/S5655/python/rule.adoc (hunk 2 above)
Comment (joke1196) [reply rules/S5655/python/rule.adoc:5]: On this one, I mean all the type of all the arguments. So maybe arguments' type would be better?

Comment (jean-jimbo-sonarsource) [reply rules/S5655/python/rule.adoc]: No, it doesn't. It's a contraction of "does not check types of any/all arguments when..."

Comment (joke1196) [reply rules/S5655/python/rule.adoc]: Perfect thank you. I made the change.

**Review by jean-jimbo-sonarsource — APPROVED** [review]

**Pull request merged**